WORKFLOW.md

```diff
@@ -59,3 +59,74 @@ PR and proposal was approved by "vast majority".
 
 You are welcome to propose changes to this workflow by opening an
 [issue](https://github.com/fortran-lang/stdlib/issues).
+
+
+## Build system
+
+This project supports two build systems right now, CMake and make.
+Eventually, stdlib will be using the Fortran package manager
+([fpm](https://github.com/fortran-lang/fpm)) as build system as well.
+The effort of supporting fpm is tracked in issue
+[#279](https://github.com/fortran-lang/stdlib/issues/279).
+
+
+### CMake build files
+
+The build files for CMake allow both in-tree, *i.e.* build artifacts share
+the same tree as the source files, and out-of-tree builds, *i.e.* build artifacts
+exist in a separate directory tree.
+Both build types are explicitly supported and tested, the latter strategy
+is recommended for local development.
+
+Sources for the main library target are added in ``src/CMakeLists.txt``
+relative to their the library target, *i.e.* no absolute paths are required.
+
+To add tests the macro ``ADDTEST`` should be used in instead of the CMake function
+``add_test``, it hides creation of the executable target, linking against the
+main library target and registering the test.
+The tests themselves are defined as standalone executables in the subdirectories
+in ``src/tests``, a new subdirectory with tests has to registred in
+``src/tests/CMakeLists.txt``.
+
+The source tree should be considered read-only, references to ``PROJECT_SOURCE_DIR``
+and ``CMAKE_CURRENT_SOURCE_DIR`` should only be used for accessing source files,
+never to write build outputs, use ``PROJECT_BINARY_DIR`` and ``CMAKE_CURRENT_BINARY_DIR``
+to write build artifacts instead.
+To fully support in-tree builds, build artifacts must never have the same name as
+source files to avoid accidentally overwriting them, *e.g.* when preprocessing or
+configuring a file.
+
+The ``CMAKE_INSTALL_PREFIX`` should only be written to on install, never in the build
+process. To install generated files, create an build output in the build tree and
+install it with the correct ``install`` function.
```

Comment:
For a user who is not familiar with `cmake`, "a correct ``install`` function" might be a vague terminology.

Reply:
Since there is only one install function in CMake with dozen of different invocations and arguments this will be hard to make more concrete. Point is to discourage using `configure_file` or similar to generate files in the install prefix on configuration or build steps, like I have seen in some CMake based projects already.

Maybe removing the correct is better?
```suggestion
install it with the ``install`` function.
```